shared/actions/json/fs.json

```diff
@@ -13,6 +13,23 @@
     "sortSetting": {
       "path": "Types.Path",
       "sortSetting": "Types.SortSetting"
+    },
+    "download": {
+      "path": "Types.Path",
+      "localPath?": "string"
+    },
+    "downloadStarted": {
+      "key": "string",
```

Comment:
unclear if we want this key or we just want to talk about paths... we can change this later but worth talking about.

Reply:
We need to:

1) differentiate download from upload;

2) be able to download the same item multiple times and auto rename them.

For 1) I guess the alternative would be using different types for upload and download, but they really share a lot of stuff. For 2) I think we can just use local path instead of KBFS path. I feel using a unique for each transfer is clearer and more flexible. WDYT?

Reply:
k well its not clear to me all the use cases so we can keep it for now.
1 could be just a boolean and 2 could be not allowed. if you download something we can just detect it and not do it

Reply:
> 2 could be not allowed. if you download something we can just detect it and not do it

If the file already exists in the destination, we can't just overwrite it, and we don't really know if it's something else or a previous download of the same file, or if user has downloaded the file, modified, and wanted a original version so they downloaded again. It's kind of annoying to require user to rename the file in the Downloads folder in this case. Wouldn't it be nicer to auto-rename them by appending a number suffix as of now?